dist/CMakeLists.txt

```diff
@@ -1,9 +1,7 @@
-find_package(Qt5Widgets ${requiredQtVersion}) # to conditionally install desktop-related files
-
-if (APPLE)
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
     add_subdirectory(mac)
-elseif (UNIX)
+elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
```

Comment:
@glassez 

Might be. But I haven't tested, and neither has anyone else (that I know of). The lazy way would be to replace all instances of `if(CMAKE_SYSTEM_NAME MATCHES "Linux")` with `if(UNIX AND (NOT APPLE))` and hope for the best, but this doesn't actually guarantee anything - we might be misleading the users into thinking that the support is in good standing. I think if someone wants to make it work in those systems, they can test and submit patches later with the required changes.

Reply:
>But I haven't tested, and neither has anyone else (that I know of).

AFAIK, qBittorrent is used at least on FreeBSD. But even if someone does not know this clearly, it does not mean that it does not exist...
There are no such big differences between different \*nix systems (and there is support in the source code), so it can be configured and built there even if no one has explicitly taken care of it.
You don't have to break it here. It will still be possible to run this script on the mentioned systems, and it will be a shame if it fails just because of such changes.
@Chocobo1, what do you think?

Reply:
>@Chocobo1, what do you think?

Unless proven to work, I would choose the conservative way to avoid unnecessary breaking.

Reply:
@Chocobo1 @glassez 

> Unless proven to work reliably, I would choose the conservative way to avoid unnecessary breaking.

`CMAKE_SYSTEM_NAME MATCHES "Linux"` works reliably to mean "Linux only".

I took a closer look at the scripts, and figured out that even now, the build should work on the BSDs. Currently, there are 4 places in the code that could be changed from that to `UNIX AND NOT APPLE AND NOT CYGWIN`:

- finding libtorrent through pkg-config (finding via cmake config files should work fine anyway)
- the script in the `dist folder`, that only handles the installation of man pages, icons, etc (thus, not relevant for building only; are the current installation directories for these ancillary files also correct for the BSDs?)
- the addition of the `_DEFAULT_SOURCE` compile definition to all targets
- the availability of the DBus and SystemD features (is it relevant to the BSDs?)

The latter item can remain disabled on non-Linux for now, and the others enabled without affecting anything else. Then, at least a basic build without the `SYSTEMD` and `DBUS` features should work. The installation will also work; if any of the installation directories need to be changed, some BSD user can later submit a patch for that.

I'll make the change.